qdev_wrappers/customised_instruments/heterodyne_source.py

```diff
@@ -0,0 +1,128 @@
+from qcodes.instrument.base import Instrument
+from qcodes.utils import validators as vals
+import logging
+from qdev_wrappers.parameters import DelegateParameter
+
+# TODO: name help pleaaase!
+# TODO: should the cavity and local os actually be parameters?
+
+
+class HeterodyneSource(Instrument):
+    """
+    This driver is a wrapper used to set two SGS100A microwave sources
+    for use as part of a lock in amplifier together with an Alazar
+    acquisition card and assumingthe physical setup including mixing
+    and a low pass filter.
+    """
+
+    def __init__(self, name, cavity=None, localos=None,
+                 demodulation_frequency=None):
+        self._cavity = cavity
+        self._localos = localos
+        if cavity is None or localos is None:
+            raise RuntimeError('must initialise with microwave sources '
+                               'for cavity and local oscillator')
+        super().__init__(name)
+
+        self.add_parameter(name='frequency',
+                           set_cmd=self._set_drive_frequency,
+                           get_cmd=self._cavity.frequency,
+                           vals=vals.Numbers(1e6, 20e9))
```

Comment:
I think it would be handy with a docstring stating explicitly how this sets the freqs of lo and cavity.
Something like:

docstring='Sets cavity frequency to the input frequency and LO frequency to input frequency + demodulation_frequency`

Reply:
Same applies below